### PR TITLE
fix required follows optional parameter deprecation draftstatementservice

### PR DESCRIPTION
**Pull request by mussbach**
This PR fixes the php 8 deprecation and places optional parameters behind mandatory ones

### Linked PRs (optional)
run phpstan

- Other PR1 #{PR-number1}
- Other PR2 #{PR-number2}
-->

### PR Checklist

Delete the checkbox if it doesn't apply/isn't necessary.

- [ ] Tests updated/created
- [ ] Update documentation
- [ ] Link all relevant tickets
- [ ] Move the tickets on the board accordingly

### CODE REVIEW

File: demosplan/DemosPlanStatementBundle/Logic/DraftStatementService.php [hunk 1]
```diff
@@ -1384,16 +1384,16 @@ protected function createSingleDocumentVersion(SingleDocument $singleDocument)
...
      * @param array|null  $filters
      * @param array|null  $sort
      * @param string|null $manualSortScope
      * @param array       $aggregation     Elasticsearch aggregation converted to legacy
-     * @param string      $procedureId
      *
      * @internal param array $filter
      */
-    protected function toLegacyResult($list, $search = '', $filters = [], $sort = [], $manualSortScope = null, $aggregation = [], $procedureId): DraftStatementResult
+    protected function toLegacyResult($list, $procedureId, $search = '', $filters = [], $sort = [], $manualSortScope = null, $aggregation = []): DraftStatementResult
```
Comment (breierDemos) [inline demosplan/DemosPlanStatementBundle/Logic/DraftStatementService.php:1396]:
```suggestion
    protected function toLegacyResult($list, string $procedureId, $search = '', $filters = [], $sort = [], string $manualSortScope = null, $aggregation = []): DraftStatementResult
```

File: demosplan/DemosPlanStatementBundle/Logic/DraftStatementService.php [hunk 2]
```diff
@@ -1384,16 +1384,16 @@ protected function createSingleDocumentVersion(SingleDocument $singleDocument)
      * Convert Result to Legacy.
      *
      * @param array       $list
+     * @param string      $procedureId
```
Comment (breierDemos) [inline demosplan/DemosPlanStatementBundle/Logic/DraftStatementService.php:1387]:
```suggestion
```

File: demosplan/DemosPlanStatementBundle/Logic/DraftStatementService.php [hunk 3]
```diff
@@ -1384,16 +1384,16 @@ protected function createSingleDocumentVersion(SingleDocument $singleDocument)
...
      * @param string      $search
      * @param array|null  $filters
      * @param array|null  $sort
      * @param string|null $manualSortScope
```
Comment (breierDemos) [inline demosplan/DemosPlanStatementBundle/Logic/DraftStatementService.php:1391]:
```suggestion
```

**Review by breierDemos — CHANGES_REQUESTED** [review]
Some suggestions

File: demosplan/DemosPlanStatementBundle/Logic/DraftStatementService.php (hunk 2 above)
Comment (mussbach) [reply demosplan/DemosPlanStatementBundle/Logic/DraftStatementService.php:1387]: See above, without defining the param types this should remain

File: demosplan/DemosPlanStatementBundle/Logic/DraftStatementService.php (hunk 3 above)
Comment (mussbach) [reply demosplan/DemosPlanStatementBundle/Logic/DraftStatementService.php:1391]: See above, without defining the param types this should remain

**Review by breierDemos — APPROVED** [review]

**Pull request merged**